### DISCUSSION TITLE
You are running the esm-bundler build of vue-i18n. It is recommended to configure your bundler to explicitly replace feature flag globals with boolean literals to get proper tree-shaking in the final bundle.

### DISCUSSION THREAD

**Discussion by dybxin** [intlify/vue-i18n#573] [Q&A]
You are running the esm-bundler build of vue-i18n. It is recommended to configure your bundler to explicitly replace feature flag globals with boolean literals to get proper tree-shaking in the final bundle.

**Comment by Liwoj** [comment]
[Reduce bundle size with feature build flags](https://vue-i18n.intlify.dev/guide/advanced/optimization.html#reduce-bundle-size-with-feature-build-flags)

**Comment by kazupon** [comment]
You can do this with the links in the comments above.

**Comment by Liwoj** [comment]
And here is example how to do it in Vue CLI app (if you are NOT using `vue-cli-plugin-i18n`)

```
// vue.config.js
const webpack = require('webpack')

module.exports = {
  chainWebpack: (config) => {
      config.plugin('vue-i18n-feature-flags').use(webpack.DefinePlugin, [{
        __VUE_I18N_LEGACY_API__: 'false',
        __VUE_I18N_FULL_INSTALL__: 'true',
        __VUE_I18N_PROD_DEVTOOLS__: 'false'
      }])
  }
}
```

Note that in current release branch (9.1.x) the warning will show even if you use config above (because #514 was not merged into 9.1.x branch)
